packages/flutter/lib/src/widgets/inherited_theme.dart

```diff
@@ -0,0 +1,146 @@
+///               // because its ancestors are above MyApp's home.
+///               return InheritedTheme.captureAll(context, Container(
+///                 alignment: Alignment.center,
+///                 color: Theme.of(context).colorScheme.surface,
+///                 child: Text('Hello World'),
+///               ));
+///             },
+///           ),
+///         );
+///       },
+///       child: Center(child: Text('Tap Here')),
+///     );
+///   }
+/// }
+///
+/// class MyApp extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return MaterialApp(
+///       home: Scaffold(
+///         // Override the DefaultTextStyle defined by the Scaffold.
+///         // Descendant widgets will inherit this big blue text style.
+///         body: DefaultTextStyle(
+///           style: TextStyle(fontSize: 48, color: Colors.blue),
+///           child: MyAppBody(),
+///         ),
+///       ),
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+abstract class InheritedTheme extends InheritedWidget {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const InheritedTheme({
+    Key key,
+    @required Widget child,
+  }) : super(key: key, child: child);
+
+  /// Return a copy of this inherited theme with the specified [child].
+  ///
+  /// If the identical inherited theme is already visible from [context] then
+  /// just return the [child].
+  ///
+  /// This implementation for [TooltipTheme] is typical:
+  /// ```dart
+  /// Widget wrap(BuildContext context, Widget child) {
+  ///   final TooltipTheme ancestorTheme = context.ancestorWidgetOfExactType(TooltipTheme);
+  ///   return identical(this, ancestorTheme) ? child : TooltipTheme(data: data, child: child);
+  /// }
+  /// ```
+  Widget wrap(BuildContext context, Widget child);
+
+  /// Returns a widget that will [wrap] child in all of the inherited themes
+  /// which are visible from [context].
+  static Widget captureAll(BuildContext context, Widget child) {
+    assert(child != null);
+    assert(context != null);
+
+    final List<InheritedTheme> themes = <InheritedTheme>[];
+    context.visitAncestorElements((Element ancestor) {
+      if (ancestor is InheritedElement && ancestor.widget is InheritedTheme) {
+        final InheritedTheme theme = ancestor.widget;
+        themes.add(theme);
+      }
+      return true;
+    });
+
+    return _CaptureAll(themes: themes, child: child);
+  }
+}
+
+class _CaptureAll extends StatelessWidget {
+  const _CaptureAll({
+    Key key,
+    @required this.themes,
+    @required this.child
+  }) : assert(themes != null), assert(child != null), super(key: key);
+
+  final List<InheritedTheme> themes;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    Widget wrappedChild = Builder(builder: (BuildContext context) => child);
```

Comment:
I don't quite understand the purpose of having this `Builder`, could you shed some light on this?

Reply:
It wasn't needed; thanks for catching that.